realm/index.d.ts

```diff
@@ -107,24 +107,24 @@ declare namespace Realm {
          * @param  {any[]} ...arg
          * @returns Results
          */
-        filtered(query: string, ...arg: any[]): Results;
+        filtered(query: string, ...arg: any[]): Results<T & Object>;
```

Comment:
Ignore the above - I just realized you were referring to `Realm.Object`. Can you please make that explicit?